components/Login/RegisterForm.tsx

```diff
@@ -70,304 +68,291 @@ export default function RegisterForm() {
     setShowConfirmPassword(!showConfirmPassword);
   };
 
-  const checkIfPasswordsMatch = (password, confirmPassword) => {
+  const checkIfPasswordsMatch = (
+    password: string,
+    confirmPassword: string,
+  ): void => {
     if (password === confirmPassword) {
       setConfirmPasswordAlert("");
     } else {
-      // The passwords do not match
       setConfirmPasswordAlert("Os campos de senha precisam ser iguais");
     }
   };
 
-  // TODO: This function should take into consideration
-  // that alerts might be empty when input is yet to be given
-  /**
-   * Function for validating all input fields' content
-   */
-  function validateInput() {
-    const validationPassed =
-      nameAlert === "" &&
-      emailAlert === "" &&
-      firstName != "" &&
-      lastName != "" &&
-      email != "" &&
-      passwordLengthValid &&
-      passwordContainsLetter &&
-      confirmPasswordAlert === "";
-
-    setIsAllInputValid(validationPassed);
-  }
-
   /**
    * Function for registering a new user in the database
-   * @param {String} firstName
-   * @param {String} lastName
-   * @param {String} email
-   * @param {String} password
    */
-  async function register() {
-    validateInput(firstName, email, password);
-
+  const register = (name: string, email: string, password: string): void => {
     if (!isAllInputValid) {
       return;
     }
 
-    //Sanitize names to removes spaces infront and after
-    const sanitizedFirstName = firstName.trim();
-    const sanitizedLastName = lastName.trim();
-    const lowerCaseEmail = email.toLowerCase();
-
     const obj = {
-      firstName: sanitizedFirstName,
-      lastName: sanitizedLastName,
-      email: lowerCaseEmail,
+      firstName: name.trim(),
+      // TODO: once backend is setup and api call are handled properly, the last name should be removed
+      lastName: "TODOFIX",
+      email: email.toLowerCase(),
       password: password,
     };
 
     try {
-      await registerUser(obj)
-        .then(async function (response) {
-          // Save user info in storage
-          // TODO: Refactor backend to get the same response as on login
-          const userInfo = {
-            id: response.baseUser._id,
-            ...response.baseUser,
-          };
-          await setUserInfo(userInfo);
-        })
-        .then(async function () {
+      registerUser(obj)
+        .then(
+          async (response: { baseUser: { _id: string; user: UserInfo } }) => {
+            // Save user info in storage
+            // TODO: Refactor backend to get the same response as on login
+            const userInfo: UserInfo = {
+              id: response.baseUser._id,
+              firstName: response.baseUser.user.firstName,
+              lastName: response.baseUser.user.lastName,
+              email: response.baseUser.user.email,
+            };
+            await setUserInfo(userInfo);
+          },
+        )
+        .then(async () => {
```

Comment:
Prefer `try/catch` with `await`. Nested, repetitive chained callbacks are difficult to read and work with.